### PR TITLE
upgpkg(x11/weston): 14.0.0

### PR DESCRIPTION
**Pull request by 2096779623**

### CODE REVIEW

File: x11-packages/weston/0008-drop-tests.patch [hunk 1]
```diff
@@ -1,11 +1,12 @@
...
- subdir('data')
- subdir('man')
- subdir('pam')
+--- a/meson.build
++++ b/meson.build
+@@ -194,9 +194,6 @@
+ subdir('pipewire')
+ subdir('clients')
+ subdir('wcap')
+-if get_option('tests')
```
Comment (Biswa96) [inline x11-packages/weston/0008-drop-tests.patch:7]: Could this be disabled with `-Dtests=false` meson option?

File: x11-packages/weston/bump-neatvnc.patch [hunk 2]
```diff
@@ -1,10 +1,10 @@
 --- a/libweston/backend-vnc/meson.build
 +++ b/libweston/backend-vnc/meson.build
-@@ -3,7 +3,7 @@ if not get_option('backend-vnc')
+@@ -3,7 +3,7 @@
  endif
  
  config_h.set('BUILD_VNC_COMPOSITOR', '1')
--dep_neatvnc = dependency('neatvnc', version: ['>= 0.7.0', '< 0.8.0'], required: false, fallback: ['neatvnc', 'neatvnc_dep'])
+-dep_neatvnc = dependency('neatvnc', version: ['>= 0.7.0', '< 0.9.0'], required: false, fallback: ['neatvnc', 'neatvnc_dep'])
 +dep_neatvnc = dependency('neatvnc', version: ['>= 0.7.0'], required: false, fallback: ['neatvnc', 'neatvnc_dep'])
```
Comment (Biswa96) [inline x11-packages/weston/bump-neatvnc.patch:8]: This patch may not be required. neatvnc is version 0.8.1 which is in the required version range.

File: x11-packages/weston/0002-drop-deps-libinput.patch [hunk 3]
```diff
@@ -48,70 +11,116 @@
...
+-
 +#else
 +	weston_log("error: drm_backend is not avaiable on Android.\n");
 +	abort();
 +#endif
-+
  	return 0;
  }
  
+diff -uNr weston-14.0.0/frontend/meson.build weston-14.0.0.mod/frontend/meson.build
```
Comment (licy183) [inline x11-packages/weston/0002-drop-deps-libinput.patch:86]: This block can be moved to the beginning of thie file, which can reduce the diff between the two commits.

File: x11-packages/weston/0010-drop-deps-gbm.patch [hunk 4]
```diff
@@ -0,0 +1,230 @@
+diff -uNr weston-14.0.0/libweston/libweston-internal.h weston-14.0.0.mod/libweston/libweston-internal.h
+--- weston-14.0.0/libweston/libweston-internal.h	2024-09-04 16:53:50.000000000 +0800
++++ weston-14.0.0.mod/libweston/libweston-internal.h	2024-09-07 20:40:27.532426816 +0800
+@@ -142,26 +142,6 @@
+ 	void (*remove_renderbuffer_dmabuf)(struct weston_output *output,
+ 					   struct weston_renderbuffer *renderbuffer);
+ 
+-	/* Allocate a DMABUF that can be imported as renderbuffer
```
Comment (licy183) [inline x11-packages/weston/0010-drop-deps-gbm.patch:8]: It is better to use `__ANDROID__` guard rather than remove this code block.

**Review by licy183 — APPROVED** [review]

**Pull request merged**